dbg.h

```diff
@@ -79,6 +81,27 @@ inline bool isColorizedOutputEnabled() {
 }
 #endif
 
+namespace timestamp {
+
+// Provide a timestamp at the precision of microsecond.
+// Not exactly the current microsecond yet, still a useful reference.
+
+struct timestamp_t {
```

Comment:
I suggest we remove the `namespace timestamp` and simply name this type `timestamp` (it's still in the `dbg` namespace).

Reply:
I just thought I could use this namespace to distinguish the relevant code for the timestamp, but I can remove it if unnecessary.